packages/contracts/deploy/007-OVM_L1CrossDomainMessenger.deploy.ts

```diff
@@ -1,56 +1,41 @@
 /* Imports: External */
 import { DeployFunction } from 'hardhat-deploy/dist/types'
+import { hexStringEquals } from '@eth-optimism/core-utils'
 
 /* Imports: Internal */
-import { getDeployedContract } from '../src/hardhat-deploy-ethers'
+import {
+  getDeployedContract,
+  deployAndRegister,
+  waitUntilTrue,
+} from '../src/hardhat-deploy-ethers'
 
 const deployFn: DeployFunction = async (hre) => {
-  const { deploy } = hre.deployments
-  const { deployer } = await hre.getNamedAccounts()
-
   const Lib_AddressManager = await getDeployedContract(
     hre,
-    'Lib_AddressManager',
-    {
-      signerOrProvider: deployer,
-    }
+    'Lib_AddressManager'
   )
 
-  const result = await deploy('L1CrossDomainMessenger', {
-    from: deployer,
+  await deployAndRegister({
+    hre,
+    name: 'L1CrossDomainMessenger',
     args: [],
-    log: true,
+    postDeployAction: async (contract) => {
+      // Theoretically it's not necessary to initialize this contract since it sits behind
+      // a proxy. However, it's best practice to initialize it anyway just in case there's
+      // some unknown security hole. It also prevents another user from appearing like an
+      // official address because it managed to call the initialization function.
+      console.log(`Initializing L1CrossDomainMessenger...`)
+      await contract.initialize(Lib_AddressManager.address)
```

Comment:
Need to make sure that we handle the case where this transaction gets stuck in the mempool.

Reply:
Not required for this PR